src/components/screens/vacation-requests/vacation-requests-table/vacation-requests-table-toolbar/vacation-requests-table-toolbar.tsx

```diff
@@ -0,0 +1,188 @@
+import { Add, Cancel, Edit } from "@mui/icons-material";
+import { Box, Collapse, Grid, Typography, styled } from "@mui/material";
+import { useEffect, useState } from "react";
+import ToolbarForm from "./toolbar-form/toolbar-form";
+import { GridRowId } from "@mui/x-data-grid";
+import { DataGridRow, ToolbarFormModes, VacationData } from "../../../../../types";
+import ToolbarDeleteButton from "./toolbar-delete-button";
+import FormToggleButton from "./toolbar-form-toggle-button";
+import ConfirmationHandler from "../../../../contexts/confirmation-handler";
+import strings from "../../../../../localization/strings";
+import { getToolbarTitle } from "../../../../../utils/toolbar-utils";
+import { VacationRequestStatuses } from "../../../../../generated/client";
+import UpdateStatusButton from "./toolbar-update-status-button";
+
+/**
+ * Component properties
+ */
+interface Props {
+  deleteVacationRequests: (selectedRowIds: GridRowId[], rows: DataGridRow[]) => Promise<void>;
+  createVacationRequest: (vacationData: VacationData) => Promise<void>;
+  updateVacationRequest: (
+    vacationData: VacationData,
+    vacationRequestId: string | undefined
+  ) => Promise<void>;
+  updateVacationRequestStatuses: (
+    newStatus: VacationRequestStatuses,
+    selectedRowIds: GridRowId[]
+  ) => Promise<void>;
+  setFormOpen: (formOpen: boolean) => void;
+  formOpen: boolean;
+  selectedRowIds: GridRowId[];
+  rows: DataGridRow[];
+}
+/**
+ * Table toolbar component
+ *
+ * @param props component properties
+ */
+const TableToolbar = ({
+  deleteVacationRequests,
+  createVacationRequest,
+  updateVacationRequest,
+  updateVacationRequestStatuses,
+  setFormOpen,
+  formOpen,
+  selectedRowIds,
+  rows
+}: Props) => {
+  const [toolbarOpen, setToolbarOpen] = useState<boolean>(false);
```

Comment:
You are still explicitly stating the type as boolean here, but this is not necessary in this case

Reply:
removed type